src/config.cpp

```diff
@@ -129,6 +151,27 @@ std::string toAbsolute(const fs::path& base, const fs::path& path) {
     return absolute.lexically_normal().string();
 }
 
+template <typename Type>
+bool is_in(const Type& value, const std::initializer_list<Type>& lst) {
+    return (std::find(std::begin(lst), std::end(lst), value) != std::end(lst));
+}
+
+template <typename Type>
+void checkValidField(const Type& field, const std::initializer_list<Type>& possibleValues) {
+    if (!is_in<Type>(field, possibleValues)) {
+        throw SonataError(fmt::format("Field '{}' not supported ('{}') possible",
+                                      field,
+                                      fmt::join(possibleValues, ", ")));
+    }
+}
+
+template <typename Type>
+void checkValidEnum(const Type& field) {
```

Comment:
Shall I add something here so the error message is more descriptive? (i guess we would need an enum.toString...)

Reply:
Yes, sounds better, message like in `checkValidField`. In fact, I just notice, where is `checkValidField` used ?

Reply:
@WeinaJi i noticed also that they are not used and will remove them.
@mgeplf do you think is worth add a string member for the enum fields in order to get a better error message in checkValidEnum() method?
```
Type type;
string type_name;
...
```

Reply:
> do you think is worth add a string member for the enum fields in order to get a better error message in checkValidEnum() method?

not sure what you mean by a `string member`?

Having a hint about which key is being checked would be a minimum, in my opinion.

One option would be to logic to `parseMandatory` / `parseOptional` that `std::enable_if` by `std::is_enum` that adds better error messages?  That way, the erroneous value is easily accessible, and it can be included in the error?

Reply:
Nevermind, I think printing the key is enough, I feel adding the string value in the error message is too complicated.
You can go ahead and merge whenever you feel!